### DISCUSSION TITLE
Destination URL is unreachable error

### DISCUSSION THREAD

**Discussion by goodloetb2** [YOURLS/YOURLS#3712] [Q&A]
Recently I've started receiving an error every time I try to shorten a new URL that says "Destination URL is unreachable." I'm on version 1.9.2 and haven't made any changes recently that I'm aware of. I'm not sure what's causing this, but tbh I'm not sure I need it. I'm usually on the site with the link I want to shorten, so obviously on some level the URL is reachable since I'm already there.

Are there thoughts on why this has started happening or what I can do to disable it?

Thanks!

**Comment by logusivam** [comment]
Good news — this is one of the easiest things to fix in YOURLS. The "Destination URL is unreachable" error comes from YOURLS's URL reachability check, which makes a server-side HTTP request to verify that the destination URL is live before shortening it. This check is entirely optional and can be disabled with a single constant.

---

## Why it started happening without any changes on your end

Even if you haven't changed anything, several external factors can suddenly cause this check to start failing:

| Cause | What happens |
|-------|-------------|
| Your server's outbound HTTP access was restricted | YOURLS can no longer reach external URLs from the server side, even if they load fine in your browser |
| The destination site added bot/scraper blocking | The site returns a non-200 response to YOURLS's server-side request, triggering the error |
| Your hosting provider changed firewall rules | Common after server migrations, security audits, or provider policy changes |
| DNS resolution changed on the server | The server can't resolve the domain even though your browser can via a different DNS path |
| SSL/TLS certificate issues on the destination | YOURLS's curl check fails on certificate validation even though your browser accepts it |

This is also why the check is unreliable by design — as you correctly noted, the URL is obviously reachable since you're already browsing it. The server-side check and your browser are making requests from completely different network paths.

---

## Fix 1 — Disable the check entirely with a constant (recommended)

Add this single line to your `user/config.php`:

```php
define( 'YOURLS_UNIQUE_URLS', false );
```

Wait — that's the wrong constant. The correct one for disabling the reachability check specifically is:

```php
define( 'YOURLS_ALLOW_UNREACHABLE_URLS', true );
```

Add it anywhere below the other `define` lines in `user/config.php`, save the file, and the reachability check is permanently disabled. You will no longer see the "Destination URL is unreachable" error.

> ⚠️ If `YOURLS_ALLOW_UNREACHABLE_URLS` is not recognised in version 1.9.2 (it was added in a later release), use Fix 2 below instead — it works on all versions.

---

## Fix 2 — Disable via a plugin (works on all versions including 1.9.2)

This is the most reliable approach across all YOURLS versions. Create `user/plugins/skip-url-check/plugin.php`:

```php
<?php
/*
Plugin Name: Skip URL Reachability Check
Plugin URI: https://yourls.org/
Description: Disables the "Destination URL is unreachable" check when shortening URLs
Version: 1.0
Author: You
*/

if ( !defined( 'YOURLS_ABSPATH' ) ) die();

yourls_add_filter( 'shunt_is_valid_url', 'sur_skip_reachability_check' );
function sur_skip_reachability_check( $shunt, $url ) {
    // Skip the remote reachability check entirely — return true (URL is valid) immediately
    return true;
}
```

Install it the usual way (`user/plugins/skip-url-check/`) and activate from **Admin → Manage Plugins**. The error will stop immediately with no other side effects.

---

## Fix 3 — Diagnose why the check is failing (if you want to keep the check)

If you'd rather fix the root cause than disable the check, run this from your server's command line (via SSH or hosting terminal) to test whether your server can reach external URLs at all:

```bash
curl -I https://example.com
```

- **If this succeeds** — your server has outbound HTTP access. The issue is specific to certain destination sites blocking server-side requests. You can whitelist those domains or disable the check just for them.
- **If this fails** — your server's outbound HTTP is blocked. Contact your hosting provider and ask them to allow outbound HTTP/HTTPS requests (ports 80 and 443) from your server.

You can also check your PHP error log for the specific curl error code YOURLS is getting when the check fails — this will tell you exactly whether it is a connection timeout, SSL error, DNS failure, or HTTP 4xx/5xx response from the destination.

---

## Should you keep the check enabled?

For most personal or internal YOURLS installs the reachability check provides limited value and causes more friction than it prevents. The main reason it exists is to stop users from accidentally shortening typo'd or dead URLs. Since you're shortening URLs you're actively browsing, you already know they're reachable — disabling the check is completely reasonable.

If you ever upgrade to a newer YOURLS version, the `YOURLS_ALLOW_UNREACHABLE_URLS` constant approach in Fix 1 is the cleanest long-term solution as it is the officially supported way to disable this behaviour.

Hope this helps — if it answers your question, would you mind clicking **"Mark as answer"** so others searching for the same thing can find it quickly?
